specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+          label,
+          items: items.filter(
+            (defaultItem) =>
+              defaultItem.type === 'DefaultStat' &&
+              !listToUse.some(
+                ({ pageName, categoryName, itemName }) =>
+                  pageName === defaultItem.pageName &&
+                  categoryName === defaultItem.categoryName &&
+                  itemName === defaultItem.itemName
+              )
+          ),
+        }))
+        .filter(({ items }) => items.length > 0),
+    }));
+  }, [layout, defaultLayout]);
+}
+
+export function StatsPage(): JSX.Element {
+  const [customLayout, setLayout] = usePref(
+    'statistics',
+    'appearance',
+    'layout'
+  );
+  const statsSpec = useStatsSpec();
+
+  const defaultLayout = useDefaultLayout(statsSpec);
+  const layout = customLayout ?? defaultLayout;
+  const [state, setState] = React.useState<
+    | State<
+        'EditingState',
+        {
+          readonly addingItem:
+            | {
+                readonly pageIndex: number;
+                readonly categoryIndex: number;
+              }
+            | undefined;
+        }
+      >
+    | State<'DefaultState'>
+  >({ type: 'DefaultState' });
+  const isEditing = state.type === 'EditingState';
+  const isAddingItem = isEditing && state.addingItem !== undefined;
+  const [activePageIndex, setActivePageIndex] = React.useState<number>(0);
+  const defaultStatsToAdd = useDefaultStatsToAdd(
+    layout[activePageIndex],
+    defaultLayout
+  );
+  console.log(defaultStatsToAdd);
+
+  return (
+    <>
+      {isAddingItem && (
+        <AddStatDialog
+          defaultLayout={defaultStatsToAdd}
+          statsSpec={statsSpec}
+          onAdd={(item): void =>
+            isAddingItem
+              ? setLayout(
+                  replaceItem(layout, activePageIndex, {
+                    ...layout[activePageIndex],
+                    categories: replaceItem(
+                      layout[activePageIndex].categories,
+                      state.addingItem.categoryIndex,
+                      {
+                        ...layout[activePageIndex].categories[
+                          state.addingItem.categoryIndex
+                        ],
+                        items: [
+                          ...layout[activePageIndex].categories[
+                            state.addingItem.categoryIndex
+                          ].items,
+                          item,
+                        ],
+                      }
+                    ),
+                  })
+                )
+              : undefined
+          }
+          onClose={(): void =>
+            setState({
+              type: 'EditingState',
+              addingItem: undefined,
+            })
+          }
+        />
+      )}
+      <div className="h-full w-full bg-[color:var(--form-background)]">
+        <div className="mx-auto flex h-full max-w-[min(100%,var(--form-max-width))] flex-col gap-4 overflow-y-auto  p-4 ">
```

Comment:
Did you copy these class names from some other place? If so, could you reuse the container component that those places called?